### ISSUE TITLE
Improve error messages for bad templates

### ISSUE THREAD

**Issue by mblakele** [ansible/ansible#3865]
For the code and output below, is it possible to throw a meaningful error message or emit a warning for expressions like "int test - incorrect"?

```
- hosts: all[0]
  gather_facts: no

  vars:
    count: 1

  tasks:
    - name: string test
      connection: local
      debug: msg="{{ count }}"

    - name: int test - correct
      connection: local
      debug: msg="{{ count|int }}"

    - name: int test - incorrect
      connection: local
      debug: msg="{{ int(count) }}"

# end

PLAY [all[0]] ***************************************************************** 

TASK: [string test] *********************************************************** 
ok: [myhost] => {"msg": "1"}

TASK: [int test - correct] **************************************************** 
ok: [myhost] => {"msg": "1"}

TASK: [int test - incorrect] ************************************************** 
ok: [myhost] => {"msg": "{{int(count)}}"}

PLAY RECAP ******************************************************************** 
myhost : ok=3    changed=0    unreachable=0    failed=0   
```

**Comment by mpdehaan** [comment]
You may wish to turn on undefined variable detection in ansible.cfg

Since this is more of a question, feel free to stop by the mailing list.

**Comment by mblakele** [comment]
At this point in time, https://www.google.com/search?q=ansible+undefined+variable finds no documentation on that "undefined variable detection" feature. You could consider this a docs bug instead, if you like.

I've read through https://groups.google.com/forum/#!topic/ansible-project/HcelWBhYK_4 and I think I understand the issues, but I find the current implementation difficult to debug. I believe it would be better to always warn about undefined variables - or any other template problems - but provide a directive to suppress the warnings. I think this would be the right philosophy for a tool that is meant to be robust.

**Comment by mblakele** [comment]
Fixed in devel (1.3), without any extra config. This is much better.

```
TASK: [int test - incorrect] ************************************************** 
fatal: [localhost] => One or more undefined variables: 'int' is undefined
```
